File: 2433/2433.cpp
class Solution {
public:
    int findTheOriginalArrayOfPrefixXo(TreeNode* root) {
        if(!root) return 0;
        return 1+max(findTheOriginalArrayOfPrefixXo(root->left),findTheOriginalArrayOfPrefixXo(root->right));
    }
};

File: 2433/2433.go
func findTheOriginalArrayOfPrefixXo(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := findTheOriginalArrayOfPrefixXo(root.Left)
	r := findTheOriginalArrayOfPrefixXo(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}